src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        User user;
+        if (userOptional.isPresent() && (user = userOptional.get()).getPassword()
+                .equals(HashUtil.hashPassword(password, user.getSalt()))) {
+            return user;
+        }
```

Comment:
The initialization of a variable in the 'if' statement looks a little strange to me ... To simplify the the if statement, you can create a private method that will check passwords for matches.

Reply:
done